### PR TITLE
Add /log-in screen for Jetpack Cloud

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

Untitled 5

#### Testing instructions
1. Boot branch locally ( as WordPress.com, not Jetpack Cloud )
2. Open `cloud.jetpack.com` in a private window
3. Click "Authorize Jetpack Cloud"
4. Replace "WordPress.com" in the url with "calypso.localhost:3000"
5. Navigate through the flow highlighted above, making sure it appears as in the screenshots ( hint: the image is very large, so you can open it in a new tab )

### CODE REVIEW

File: client/layout/masterbar/oauth-client.jsx [hunk 1]
```diff
@@ -23,10 +28,16 @@ const DefaultOauthClientMasterbar = ( { oauth2Client } ) => (
 		<nav>
 			<ul className="masterbar__oauth-client-main-nav">
 				<li className="masterbar__oauth-client-current">
-					{ oauth2Client.icon && (
+					{ isJetpackCloudOAuth2Client( oauth2Client ) ? (
```
Comment (enejb) [inline client/layout/masterbar/oauth-client.jsx:31]: Instead of 

```
{ isJetpackCloudOAuth2Client( oauth2Client ) ? (
						<div className="masterbar__oauth-client-logo">
							<JetpackLogo full monochrome={ false } size={ 28 } />
						</div>
					) : (
						oauth2Client.icon && (
							<div className="masterbar__oauth-client-logo">
								<img src={ oauth2Client.icon } alt={ oauth2Client.title } />
							</div>
						)
					) }
```

can we define the 
```
const oauthLogo = this.getOAuthLogo( oauth2Client ); 
```

Would be a new method that retuns null or the image or Jetpack logo based on the oauth2Client that is passed in. Then we can simply things with 
```
{ oauthLogo && <div className="masterbar__oauth-client-logo">oauthLogo</div>}
```

File: client/login/wp-login/login-links.jsx [hunk 2]
```diff
@@ -230,6 +239,11 @@ export class LoginLinks extends React.Component {
 			return null;
 		}
 
+		// jetpack cloud needs to keep users in the flow
+		if ( isJetpackCloudOAuth2Client( this.props.oauth2Client ) ) {
```
Comment (enejb) [inline client/login/wp-login/login-links.jsx:243]: Here we are removing the reset your password link. 
I know its not idea but it is definitely something that users are looking for. 

Will we be adding it a back in a different PR?

Comment (enejb) [reply client/login/wp-login/login-links.jsx:243]: Maybe instead of removing the link we can send the user somewhere else?

File: client/layout/masterbar/oauth-client.jsx (hunk 1 above)
Comment (rcanepa) [reply client/layout/masterbar/oauth-client.jsx:31]: Agree with @enejb.

**Review by rcanepa — APPROVED** [review]
The flow works and looks very nice. If supporting password reset in our flow is too complicated I would leave that out in this PR.

Thanks for working on this. It's a lot of work 🙇

**Review by cleacos — COMMENTED** [review]
The flow is expected and I can see the views. Good job!

**Note:** I can see WordPress logo for a second when I navigate between the views.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3679394

Thank you @cbauerman for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3679394).
